### PR TITLE
docs: refresh session lifespan, extend samples

### PR DESCRIPTION
**Pull request by kelkarajay**
## Changes
- Refreshing overview, lifespan and refresh/extend documentation
- Drop `Session` blob in the doc and link to API reference

### CODE REVIEW

File: docs/kratos/session-management/10_session-lifespan.mdx [hunk 1]
```diff
@@ -9,21 +9,52 @@ sidebar_label: Session lifespan
...
+
+```shell
+## List all available projects
+ory list projects
+
+## Get config
+ory get identity-config {project-id} --format yaml > identity-config.yaml
+```
+
+2. Update the configuration value for session lifespan property to the desired value (Use hour (h), minute (m), second (s) to define interval, for example: 1h1m10s, 10s, 1h)
```
Comment (tomekpapiernik) [inline docs/kratos/session-management/10_session-lifespan.mdx:39]:
```suggestion
2. Update the configuration value for session lifespan property to the desired value:
```
The example that comes before the tabs will is enough :)

File: docs/kratos/session-management/10_session-lifespan.mdx [hunk 2]
```diff
@@ -9,21 +9,52 @@ sidebar_label: Session lifespan
...
+```yaml title="config.yml"
+session:
+  cookie:
+    domain: fancy-moofasa-nsuq4mdx5a.projects.oryapis.com
+    name: ory_session_fancymoofasansuq4mdx5a
+    path: /
+    persistent: false
+    same_site: Lax
+  // highlight-start
+  lifespan: 720h0m0s
```
Comment (tomekpapiernik) [inline docs/kratos/session-management/10_session-lifespan.mdx:50]:
```suggestion
  lifespan: 720h
```
Do you need the minutes and the seconds here?

File: docs/kratos/session-management/01_overview.mdx [hunk 3]
```diff
@@ -1,16 +1,19 @@
 ---
 id: overview
-title: Overview of sessions, session cookies, and session tokens in Ory
+title: Overview
```
Comment (tomekpapiernik) [inline docs/kratos/session-management/01_overview.mdx:3]:
```suggestion
title: Overview of sessions, Ory Session Cookies, and Ory Session Tokens
```

**Review by tomekpapiernik — CHANGES_REQUESTED** [review]

File: docs/kratos/session-management/01_overview.mdx [hunk 4]
```diff
@@ -27,113 +30,82 @@ For security reasons, you can't break the isolation between cookies and session
...
-        "created_at": "2021-10-14T17:55:17.886831+02:00",
-        "updated_at": "2021-10-14T18:03:14.839105+02:00"
-      }
-    ],
-    "created_at": "2021-10-14T17:55:17.886237+02:00",
-    "updated_at": "2021-10-14T17:55:17.886237+02:00"
-  }
-}
-```
+An example of an Ory session can be seen in the [Get session API](../../reference/api#tag/identity/operation/disableSession)
```
Comment (Benehiko) [inline docs/kratos/session-management/01_overview.mdx:33]: Why `disableSession` instead of `listSessions`?
https://docs-btk6arvlq-ory.vercel.app/docs/reference/api#tag/identity/operation/listSessions

File: docs/kratos/session-management/01_overview.mdx [hunk 5]
```diff
@@ -27,113 +30,82 @@ For security reasons, you can't break the isolation between cookies and session
...
 | `authenticated_at` | Indicates the time of the most recent successful authentication. This value is updated when: <ul><li>The end-user authenticates with a second factor such as TOTP</li><li>The end-user refreshes their session using the [`/self-service/login/browser`](../../reference/api.mdx#operation/initializeSelfServiceLoginFlowForBrowsers) endpoint or [`/self-service/login/api`](../../reference/api.mdx#operation/initializeSelfServiceLoginFlowWithoutBrowser) endpoint and setting `refresh` to `true`</li></ul> |
 
-## Ory Session Cookie
+### Using Ory Session Cookie
 
 An Ory Session Cookie is issued when the user signs in through a web browser. To get the session payload, send a request to the
```
Comment (Benehiko) [inline docs/kratos/session-management/01_overview.mdx:44]: Technically true, but we show a curl command below which is native using the cookie. I think maybe we should clarify or reword it to fit the narrative that cookies are meant for browser based flows (Server-side and single page).

This is also specific to which api is called e.g. `/login/browser` vs `/login/api`.

File: docs/kratos/session-management/01_overview.mdx [hunk 6]
```diff
@@ -27,113 +30,82 @@ For security reasons, you can't break the isolation between cookies and session
...
 `/sessions/whoami` endpoint:
 
+````mdx-code-block
+<Tabs groupId="sdk-console">
+<TabItem value="native" label="React">
```
[the remark this answers is not in the document]
Comment (Benehiko) [reply docs/kratos/session-management/01_overview.mdx:49]: this if fine since `Typescript` can be a file run by `node` which is server-side instead of `react` also using typescript but it is run in the browser.

File: docs/kratos/session-management/01_overview.mdx [hunk 7]
```diff
@@ -27,113 +30,82 @@ For security reasons, you can't break the isolation between cookies and session
...
 ```
 
-## Ory Session Token
+</TabItem>
+</Tabs>
+````
+
+### Using Ory Session Token
 
 An Ory Session Token is issued when the user authenticates through a client other than a web browser. To get the session payload,
```
Comment (Benehiko) [inline docs/kratos/session-management/01_overview.mdx:73]: Also technically true, but a browser application can also call `login/api`. I think we should just be more verbose and say something along the lines of Native applications such as desktop and mobile applications, terminal based applications that do not run inside a browser should use tokens instead of cookies.

File: docs/kratos/session-management/01_overview.mdx [hunk 8]
```diff
@@ -150,36 +122,53 @@ This flow is similar to [GitHub's sudo mode](https://help.github.com/en/github/a
 The session is considered privileged when its `authenticated_at` isn't older than the `privileged_session_max_age` value defined
```
Comment (Benehiko) [inline docs/kratos/session-management/01_overview.mdx:122]:
```suggestion
The session is considered privileged when its `authenticated_at` is younger than the `privileged_session_max_age` value defined
```
I don't know, rephrase this maybe? reading "isn't older than" was tough to understand. Had to read it a few times to get the grasp of it. 

Does it mean that the session privileged allows editing or disallows editing / changing username / email.. etc.?

File: docs/kratos/session-management/01_overview.mdx [hunk 9]
```diff
@@ -150,36 +122,53 @@ This flow is similar to [GitHub's sudo mode](https://help.github.com/en/github/a
...
 
 ````mdx-code-block
-import Tabs from '@theme/Tabs';
-import TabItem from '@theme/TabItem';
-
 <Tabs>
-  <TabItem value="cloud" label="Ory CLI" default>
+<TabItem value="console" label="Ory Console" default>
+
+To change the privileged session duration, go to [**Ory Console**](https://console.ory.sh/) → **Session Settings**, enter the desired value (Use hour (h), minute (m), second (s) to define interval, for example: 15m30s, 1m) in the Privileged Sessions section and click the **Save** button.
```
Comment (Benehiko) [inline docs/kratos/session-management/01_overview.mdx:133]:
```suggestion
To change the privileged session duration, go to [**Ory Console**](https://console.ory.sh/) → **Session Settings**, enter the desired value (Use hour (h), minute (m), second (s) to define interval, for example: 1h15m30s) in the Privileged Sessions section and click the **Save** button.
```

Comment (Benehiko) [reply docs/kratos/session-management/01_overview.mdx:133]: maybe we should show it can be used together `1h15m30s` since you show an example of 15m below?

File: docs/kratos/session-management/01_overview.mdx [hunk 10]
```diff
@@ -189,6 +178,8 @@ settings:
 With this configuration in place, the user can perform the actions that require a privileged session up to 15 minutes after
 signing in. When this time passes, the user must re-authenticate to access these options.
 
+### Demo
+
 Watch this video to see the flow in action:
 
 <video controls width="100%">
```
[the remark this answers is not in the document]
Comment (Benehiko) [reply docs/kratos/session-management/01_overview.mdx:185]: I don't think it's a complete waste. but I would update it to use the new Account Experience and make it explicit the intent of the demo is showing how it asks for the password again when changing the email

File: docs/kratos/session-management/10_session-lifespan.mdx [hunk 11]
```diff
@@ -80,8 +136,12 @@ session:
 </Tabs>
 ````
 
+:::note
+
 If `max-age` is set as a part of the `Set-Cookie` header, the browser deletes the cookie when it reaches the age defined in
 `max-age`.
 
 When `max-age` is not set, the browser deletes the cookie when the session ends. The session ends when the set session lifespan
 expires, or when the browser is shut down by the user.
```
Comment (Benehiko) [inline docs/kratos/session-management/10_session-lifespan.mdx:145]: Is this applicable here, since we always have a max-age right?

File: code-examples/sdk/typescript/src/session/refresh-session-native.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import { Configuration, IdentityApi } from "@ory/client"
+
+const identityApi = new IdentityApi(
+  new Configuration({
+    basePath: `https://${process.env.ORY_PROJECT_SLUG}.projects.oryapis.com`,
+    apiKey: `${process.env.ORY_ACCESS_TOKEN}`,
```
Comment (Benehiko) [inline code-examples/sdk/typescript/src/session/refresh-session-native.ts:6]: `apiKey` works here? Interesting :thinking: 
So this would automatically add the header `Authorization`: `Bearer ${token}` ?

**Review by Benehiko — CHANGES_REQUESTED** [review]

File: docs/kratos/session-management/10_session-lifespan.mdx (hunk 2 above)
Comment (kelkarajay) [reply docs/kratos/session-management/10_session-lifespan.mdx:50]: Not necessarily. I just wanted to add that explicitly there so the format is clearer to the reader (by means of repeated reinforcing 😓 )

File: docs/kratos/session-management/10_session-lifespan.mdx [hunk 13]
```diff
@@ -9,21 +9,52 @@ sidebar_label: Session lifespan
...
+ory get identity-config {project-id} --format yaml > identity-config.yaml
+```
+
+2. Update the configuration value for session lifespan property to the desired value (Use hour (h), minute (m), second (s) to define interval, for example: 1h1m10s, 10s, 1h)
+
+```yaml title="config.yml"
+session:
+  cookie:
+    domain: fancy-moofasa-nsuq4mdx5a.projects.oryapis.com
+    name: ory_session_fancymoofasansuq4mdx5a
```
[the remark this answers is not in the document]
Comment (kelkarajay) [reply docs/kratos/session-management/10_session-lifespan.mdx:45]: it isn't an actual slug. I think I copied this from an existing doc :D

File: docs/kratos/session-management/01_overview.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (tomekpapiernik) [reply docs/kratos/session-management/01_overview.mdx:185]: I don't see how showing this in motion adds value. It's a common mechanism, no need to demo it in any way.

File: docs/kratos/session-management/10_session-lifespan.mdx [hunk 14]
```diff
@@ -9,21 +9,52 @@ sidebar_label: Session lifespan
...
+  lifespan: 720h0m0s
+  // highlight-end
+```
+
+3. Update the Ory Identities configuration using the file you worked with:
 
 ```shell
-ory patch identity-config <your-project-id> \\
-  --replace '/session/lifespan="720h"' # 30 days
+ ory update identity-config {project-id} --file identity-config.yaml
```
[the remark this answers is not in the document]
Comment (kelkarajay) [reply docs/kratos/session-management/10_session-lifespan.mdx:56]: removed

File: docs/kratos/session-management/20_refresh-extend-session.mdx [hunk 15]
```diff
@@ -3,24 +3,46 @@ id: refresh-extend-sessions
...
+<TabItem value="go" label="Go" default>
 
 ```mdx-code-block
-import { getSdkUrl } from '@site/src/hooks'
-import CodeBlock from '@theme/CodeBlock'
+import goExtendSession from '!!raw-loader!../../../code-examples/sdk/go/session/refresh-session.go'
 
-<CodeBlock>{`PATCH ${getSdkUrl().url}/admin/sessions/{id}/refresh
-Authorization: Bearer {your-personal-access-token}`}</CodeBlock>
+<CodeBlock language="go">{goExtendSession}</CodeBlock>
```
[the remark this answers is not in the document]
Comment (kelkarajay) [reply docs/kratos/session-management/20_refresh-extend-session.mdx:30]: done

File: docs/kratos/session-management/20_refresh-extend-session.mdx [hunk 16]
```diff
@@ -3,24 +3,46 @@ id: refresh-extend-sessions
...
+<CodeBlock language="ts" title="extend-session-native.ts">{nativeExtendSession}</CodeBlock>
+```
+
+</TabItem>
+</Tabs>
+````
+
 :::tip
 
 To get the Session ID, call the `/sessions/whoami` endpoint or `toSession` SDK method.
```
[the remark this answers is not in the document]
Comment (kelkarajay) [reply docs/kratos/session-management/20_refresh-extend-session.mdx:48]: fixed, also changed the text to match ory network use case

File: docs/kratos/session-management/01_overview.mdx [hunk 17]
```diff
@@ -1,16 +1,19 @@
 ---
 id: overview
-title: Overview of sessions, session cookies, and session tokens in Ory
+title: Overview
 sidebar_label: Overview
 ---
 
-# Overview
```
[the remark this answers is not in the document]
Comment (kelkarajay) [reply docs/kratos/session-management/01_overview.mdx:7]: fixed

File: docs/kratos/session-management/01_overview.mdx (hunk 4 above)
Comment (kelkarajay) [reply docs/kratos/session-management/01_overview.mdx]: fixed to `getSession`

File: docs/kratos/session-management/01_overview.mdx (hunk 8 above)
Comment (kelkarajay) [reply docs/kratos/session-management/01_overview.mdx:124]: Yeah, privileged sessions allow/enable editing fields like email

File: docs/kratos/session-management/01_overview.mdx (hunk 9 above)
Comment (kelkarajay) [reply docs/kratos/session-management/01_overview.mdx]: added this example

File: code-examples/sdk/typescript/src/session/refresh-session-native.ts (hunk 12 above)
Comment (kelkarajay) [reply code-examples/sdk/typescript/src/session/refresh-session-native.ts:6]: correction - this should be `accessToken`. Thanks!

Comment (kelkarajay) [reply code-examples/sdk/typescript/src/session/refresh-session-native.ts]: fixed https://github.com/ory/docs/pull/1185/commits/cf9380510ad32faad8b2af0142a5ca2eaf9fe119

File: docs/kratos/session-management/10_session-lifespan.mdx (hunk 11 above)
Comment (kelkarajay) [reply docs/kratos/session-management/10_session-lifespan.mdx:128]: Apparently not, when the persistent cookie configuration is turned off, it isn't - 

```go
	if s.r.Config().SessionPersistentCookie(ctx) {
		if session.ExpiresAt.IsZero() {
			cookie.Options.MaxAge = int(s.r.Config().SessionLifespan(ctx).Seconds())
		} else {
			cookie.Options.MaxAge = int(time.Until(session.ExpiresAt).Seconds())
		}
	}
```

File: docs/kratos/session-management/01_overview.mdx [hunk 18]
```diff
@@ -27,113 +30,82 @@ For security reasons, you can't break the isolation between cookies and session
...
-        "updated_at": "2021-10-14T18:03:14.839105+02:00"
-      }
-    ],
-    "created_at": "2021-10-14T17:55:17.886237+02:00",
-    "updated_at": "2021-10-14T17:55:17.886237+02:00"
-  }
-}
-```
+An example of an Ory session can be seen in the [Get session API](../../reference/api#tag/identity/operation/disableSession)
+response.
```
[the remark this answers is not in the document]
Comment (kelkarajay) [reply docs/kratos/session-management/01_overview.mdx]: added

File: docs/kratos/session-management/01_overview.mdx (hunk 5 above)
Comment (kelkarajay) [reply docs/kratos/session-management/01_overview.mdx:126]: The curl command example is only for developer simplicity to view on command line. I can add a note in about the appropriate usage

File: docs/kratos/session-management/10_session-lifespan.mdx (hunk 11 above)
Comment (Benehiko) [reply docs/kratos/session-management/10_session-lifespan.mdx:128]: Interesting, so inside of the project configuration this is configurable to Ory Network users?

**Review by Benehiko — APPROVED** [review]

File: docs/kratos/session-management/10_session-lifespan.mdx [hunk 19]
```diff
@@ -9,21 +9,52 @@ sidebar_label: Session lifespan
...
-Run this command to adjust the session lifespan to 720 hours (30 days):
+In the configuration, the session lifespan is expressed in hours, minutes and seconds combination (hour (h), minute (m), second
+(s) to define interval, for example: **1h1m10s**, **10s**, **1h**).
 
 ````mdx-code-block
 import CodeBlock from '@theme/CodeBlock'
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
 <Tabs>
```
[the remark this answers is not in the document]
Comment (kelkarajay) [reply docs/kratos/session-management/10_session-lifespan.mdx:20]: fixed

**Comment by kelkarajay** [comment]
> Interesting, so inside of the project configuration this is configurable to Ory Network users?

Yep, via cli. Project validation doesn't block updating this field

**Review by tomekpapiernik — APPROVED** [review]
